### PR TITLE
NetSuite import setting Base Pr

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/core/services/common/mapping.service.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export class MappingService {
   }
 
   getFyleFields(version?: 'v1'): Observable<FyleField[]> {
-    return this.apiService.get(`/workspaces/${this.workspaceId}/fyle/${version === 'v1' ? 'expense_fields' : 'fields'}/`, {});
+    return this.apiService.get(`/workspaces/${this.workspaceId}/fyle/${version === 'v1' ? 'expense_fields' : 'fyle_fields'}/`, {});
```
Comment (ashwin1111) [inline src/app/core/services/common/mapping.service.ts:87]: let's not change it please, this is a common service. Create a new API, just point it to same view as existing one.

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 2]
```diff
@@ -229,6 +228,97 @@ export class NetsuiteExportSettingsComponent implements OnInit {
...
+  }
+
+  private constructWarningMessage(): string {
+    let content: string = '';
+    const existingReimbursableExportType = this.exportSettings.configuration?.reimbursable_expenses_object ? this.exportSettings.configuration.reimbursable_expenses_object : 'None';
+    const existingCorporateCardExportType = this.exportSettings.configuration?.corporate_credit_card_expenses_object ? this.exportSettings.configuration.corporate_credit_card_expenses_object : 'None';
+    const updatedReimbursableExportType = this.exportSettingForm.value.reimbursableExportType ? this.exportSettingForm.value.reimbursableExportType : 'None';
+    const updatedCorporateCardExportType = this.exportSettingForm.value.creditCardExportType ? this.exportSettingForm.value.creditCardExportType : 'None';
+
+    if (this.isSingleItemizedJournalEntryAffected()) {
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:263]: this is from QBO

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 3]
```diff
@@ -229,6 +228,97 @@ export class NetsuiteExportSettingsComponent implements OnInit {
...
+
+  private isExportSettingsUpdated(): boolean {
+    return this.exportSettings.configuration.reimbursable_expenses_object !== null || this.exportSettings.configuration.corporate_credit_card_expenses_object !== null;
+  }
+
+  private isSingleItemizedJournalEntryAffected(): boolean {
+    return (this.exportSettings?.configuration?.reimbursable_expenses_object !== NetsuiteReimbursableExpensesObject.JOURNAL_ENTRY && this.exportSettingForm.value.reimbursableExportType === NetsuiteReimbursableExpensesObject.JOURNAL_ENTRY) || (this.exportSettings?.configuration?.corporate_credit_card_expenses_object !== NetSuiteCorporateCreditCardExpensesObject.JOURNAL_ENTRY && this.exportSettingForm.value.creditCardExportType === NetSuiteCorporateCreditCardExpensesObject.JOURNAL_ENTRY);
+  }
+
+  private isPaymentsSyncAffected(): boolean {
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:286]: needs update, this is from QBO

File: src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts [hunk 4]
```diff
@@ -229,6 +228,97 @@ export class NetsuiteExportSettingsComponent implements OnInit {
...
+      this.exportSettingForm.controls.creditCardExportGroup.disable();
+    } else {
+      this.cccExpenseGroupingDateOptions = this.reimbursableExpenseGroupingDateOptions.concat();
+      this.helperService.clearValidatorAndResetValue(this.exportSettingForm, 'creditCardExportGroup');
+      this.helperService.enableFormField(this.exportSettingForm, 'creditCardExportGroup');
+    }
+  }
+
+  private setupCustomWatchers(): void {
+    if (this.exportSettingForm.value.creditCardExportType && [NetSuiteCorporateCreditCardExpensesObject.EXPENSE_REPORT, NetSuiteCorporateCreditCardExpensesObject.EXPENSE_REPORT].includes(this.exportSettingForm.value.creditCardExportType)) {
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-export-settings/netsuite-export-settings.component.ts:311]: what're we doing here?

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html [hunk 5]
```diff
@@ -1 +1,77 @@
...
+                    <app-configuration-toggle-field
+                        [form]="importSettingForm"
+                        [label]="brandingContent.importCategoriesLabel + brandingConfig.brandName"
+                        [subLabel]="brandingContent.importCategoriesSubLabel"
+                        [formControllerName]="'importCategories'"
+                        [isSectionHeader]="false"
+                        [iconPath]="'arrow-tail-down'">
+                    </app-configuration-toggle-field>
+                </div>
+                <div class="tw-rounded-lg tw-border-border-tertiary tw-border tw-mt-24-px" *ngIf="isTaxGroupSyncAllowed && brandingFeatureConfig.featureFlags.importSettings.tax">
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html:27]: move this to last

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html [hunk 6]
```diff
@@ -1 +1,77 @@
...
+                    <app-configuration-toggle-field
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html:28]: should show up only for non us subs

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 7]
```diff
@@ -7,9 +23,117 @@ import { Component, OnInit } from '@angular/core';
...
+      this.mappingService.getFyleFields(),
+      this.workspaceService.getConfiguration(),
+      this.netsuiteConnectorService.getSubsidiaryMapping(),
+      this.importSettingService.getNetsuiteFields(),
+      this.netsuiteExportSettingService.getExportSettings()
+    ]).subscribe(([importSettingsResponse, fyleFieldsResponse, netsuiteConfiguration, netsuiteCredentials, netsuiteFields, exportSetting]) => {
+      this.importSettings = importSettingsResponse;
+      this.netsuiteConfiguration = netsuiteConfiguration;
+
+      if (netsuiteCredentials && netsuiteCredentials.country_name !== 'US') {
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:120]: call it subsidiaryMapping

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts [hunk 8]
```diff
@@ -7,9 +23,117 @@ import { Component, OnInit } from '@angular/core';
...
+    forkJoin([
+      this.importSettingService.getImportSettings(),
+      this.mappingService.getFyleFields(),
+      this.workspaceService.getConfiguration(),
+      this.netsuiteConnectorService.getSubsidiaryMapping(),
+      this.importSettingService.getNetsuiteFields(),
+      this.netsuiteExportSettingService.getExportSettings()
+    ]).subscribe(([importSettingsResponse, fyleFieldsResponse, netsuiteConfiguration, netsuiteCredentials, netsuiteFields, exportSetting]) => {
+      this.importSettings = importSettingsResponse;
+      this.netsuiteConfiguration = netsuiteConfiguration;
```
Comment (ashwin1111) [inline src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.ts:118]: not used anywhere, please remove api call as well

File: src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html (hunk 6 above)
Comment (NileshPant1999) [reply src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html:28]: this is handled in above div no
*ngIf="isTaxGroupSyncAllowed

Comment (ashwin1111) [reply src/app/integrations/netsuite/netsuite-shared/netsuite-import-settings/netsuite-import-settings.component.html]: cool, can ignore this comment

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**